backend/onyx/tools/tool_implementations/fetch_single_file/fetch_single_file_tool.py

```
@@ -0,0 +1,900 @@
+
+        for result in results:
+            doc_id = result.metadata.get("document_id", "unknown")
+
+            # If we haven't seen this document ID, or this result has higher confidence
+            if doc_id not in document_map or self._get_relevance_score(
+                result
+            ) > self._get_relevance_score(document_map[doc_id]):
+                document_map[doc_id] = result
+
+        # Convert back to list and sort by relevance score
+        unique_results = list(document_map.values())
+        unique_results.sort(key=lambda x: self._get_relevance_score(x), reverse=True)
+
+        logger.info(
+            f"Deduplicated {len(results)} results to {len(unique_results)} unique documents"
+        )
+        return unique_results
+
+    def _get_relevance_score(self, result: DocumentResult) -> int:
+        """Helper to get the relevance score (defaults to 0 if not set)"""
+        return result.relevance_score if result.relevance_score is not None else 0
+
+    def _search_internal_docs(self, description: str) -> list[DocumentResult]:
+        """Search internal documents using direct VespaIndex retrieval"""
+        logger.info(f"Searching internal docs for: {description}")
+
+        try:
+            # Get search settings and document index
+            get_current_search_settings(self.db_session)
+            document_index = get_current_primary_default_document_index(self.db_session)
+
+            if not isinstance(document_index, VespaIndex):
+                logger.warning(
+                    "Document index is not VespaIndex, skipping internal search"
+                )
+                return []
+
+            # Build access filters for the user
+            user = User(id=self.user_id) if self.user_id else None
+            access_filters = (
+                build_access_filters_for_user(user=user, session=self.db_session)
+                if user
+                else None
+            )
```

just to make me less nervous, can we also raise an error if self.user_id is None and the backend auth type is anything but disabled?